docs/Authenticating-Users-via-Tokens.md

```diff
@@ -0,0 +1,89 @@
+
+The second implementation based on [Universally unique identifier](https://en.wikipedia.org/wiki/Universally_unique_identifier) (UUID) technology, allows for greater control over the data access token at the cost of additional database support. Unlike JWT based data access tokens which are self-validating, a UUID based token is a string representation of a unique number.  When a data access token is generated by the cBioPortal, a record is inserted into the `data_access_tokens` database table:
+
+
+```
+mysql> describe data_access_tokens;
++------------+--------------+------+-----+---------+-------+
+| Field      | Type         | Null | Key | Default | Extra |
++------------+--------------+------+-----+---------+-------+
+| TOKEN      | varchar(50)  | NO   | PRI | NULL    |       |
+| USERNAME   | varchar(255) | NO   | MUL | NULL    |       |
+| EXPIRATION | datetime     | NO   |     | NULL    |       |
+| CREATION   | datetime     | YES  |     | NULL    |       |
++------------+--------------+------+-----+---------+-------+
+4 rows in set (0.00 sec)
+```
+In addition to storing the token (the unique number), its creation and expiration dates along with the associated username is also stored prior to presenting it to the user.  When the data access token is presented in a subsequent web service request, the cBioPortal compares the token presented in the request against this table for validity.  If the token exists in the table and has not yet expired, the request will be satisfied.  The creation date is used by the cBioPortal when a user requests a new data access token.  Based on the server configuration, if the number of outstanding data access tokens together with this new one exceeds the number of allowable data access tokens per user, the cBioPortal may refuse to present a new one or it may invalidate the oldest one before presenting the new one to the user.
+
+### Modifying Configuration
+The following properties must be present in portal.properties in order to allow direct access to the cBioPortal web service when login is required:
+
+**Property**: dat.method (required)
+- **Description**: The desired data access token implementation.
+- **Permissible Values**: jwt, uuid, none
+- **Default Value**: none
+
+**Property**: dat.unauth_users (optional)
+- **Description**: A list of users that should not be allowed to download a data access token.
+- **Permissible Values**: A comma-delimited list of valid user names as found in the users table.
+- **Default Value**: anonymousUser
+
+**Property**: dat.ttl_seconds (required)
+- **Description**: The time in seconds between token creation and token expiration.
+- **Permissible Values**: An integer value greater than zero.
+- **Default Value**: -1
```

Comment:
What happens when then the default value is -1?  I think we should set the default value to a 30 day expiration.